features/weth/Weth.tsx

```diff
@@ -114,33 +132,199 @@ const Weth = () => {
           setError(error);
         }
       } else {
-        setError(new Error("Invalid amount"));
+        setError(new Error("Amount must be positive"));
       }
-    } else {
-      setError(new Error("Please check that you are connected."));
-    }
-  };
+    };
 
-  const handleClick = (e: FormEvent, action: ACTION_TYPE) => {
-    e.preventDefault();
+    const handleClick = (e: FormEvent, action: ACTION_TYPE) => {
+      e.preventDefault();
 
-    switch (action) {
-      case ACTION_TYPE.WRAP:
-        _submitTxn(ethAmount as string, ACTION_TYPE.WRAP);
-        break;
-      case ACTION_TYPE.UNWRAP:
-        _submitTxn(wethAmount as string, ACTION_TYPE.UNWRAP);
-        break;
-      default:
-        alert("Invalid action!");
-    }
-  };
+      switch (action) {
+        case ACTION_TYPE.WRAP:
+          _submitTxn(ethAmountToWrap, ACTION_TYPE.WRAP);
+          break;
+        case ACTION_TYPE.UNWRAP:
+          _submitTxn(wethAmountToUnwrap, ACTION_TYPE.UNWRAP);
+          break;
+        default:
+          alert("Invalid action!");
```

Comment:
Dont feel super strongly but just a question of style here: do we need to have this switch logic like this? the `_submitTxn` function seems a bit over-engineered a bit. Could you not simply deal with the two cases in two separate functions and then call the respective functions from the `wrap` and `unwrap` buttons?

Reply:
You could but is that better style to have 2 functions instead of 1 with a switch?

Reply:
well you end up having two functions anyways with `handelClick` and `_submitTxn` :)

 `handelClick` has a switch case in it and `_submitTxn` uses an if then else. if you had 1 function `wrapEth` and another `unwrapWeth` you would not need any switching or if/else logic within them and just the exact calls to the contracts with tx handling. Line 205 would then simply call `wrapEth`. 

I know it's the same as what your current logic does I just stylistically think you could do the same logic with about 50% as much code. 

Ultimately I don't really mind too much, just prefer to have simpler functions with no branching logic.

Reply:
ok I'll just create wrapEth and unwrapWeth I think that makes sense